speech/snippets/context_classes.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package snippets contains speech examples.
+package snippets
+
+import (
+	speech "cloud.google.com/go/speech/apiv1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1"
+
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// [START speech_context_classes]
+
+// contextClasses provides "hints" to the speech recognizer
+// to favour specific classes of words in the results
+func contextClasses(w io.Writer, gcsURI string) error {
+	ctx := context.Background()
+
+	client, err := speech.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewClient: %v", err)
+	}
+
+	// SpeechContext: to configure your speech_context see:
+	// https://cloud.google.com/speech-to-text/docs/reference/rpc/google.cloud.speech.v1#speechcontext
+	// Full list of supported phrases (class tokens) here:
+	// https://cloud.google.com/speech-to-text/docs/class-tokens
+	SpeechContext := &speechpb.SpeechContext{Phrases: []string{"$TIME"}}
```

Comment:
Perhaps use this comment block to explain you're demonstrating use of the "$TIME" class to favor time of day detections, as it wasn't clear on first readthru that there wasn't some kind of indirect variable substitution.